compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1911,7 +1911,7 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                     return None;
                 }
 
-                let imp = self.tcx.impl_trait_ref(def_id).unwrap();
+                let imp = self.tcx.bound_impl_trait_ref(def_id).unwrap().subst_identity();
```

Comment:
actually `skip_binder` as we again pretend like we've instantiated the `ty::Param` with inference vars